prov/util/src/util_attr.c

```diff
@@ -428,63 +428,74 @@ static int fi_resource_mgmt_level(enum fi_resource_mgmt rm_model)
 	}
 }
 
-/* If a provider supports basic registration mode it should set the FI_MR_BASIC
- * mode bit in prov_mode. Support for FI_MR_SCALABLE is indicated by not setting
- * any of OFI_MR_BASIC_MAP bits. */
-int ofi_check_mr_mode(const struct fi_provider *prov, uint32_t api_version,
-		      uint64_t user_info_caps, uint32_t prov_mode,
-		      uint32_t user_mode)
+/*
+ * Remove unneeded MR mode bits based on the requested capability bits.
+ */
+static int ofi_cap_mr_mode(uint64_t info_caps, int mr_mode)
 {
-	uint64_t prov_mode_log;
-	int ret;
-
-	if (FI_VERSION_LT(api_version, FI_VERSION(1, 5))) {
-		if (!ofi_rma_target_allowed(user_info_caps) &&
-		    !(prov_mode & FI_MR_LOCAL))
+	if (!ofi_rma_target_allowed(info_caps)) {
+		if (!(mr_mode & FI_MR_LOCAL))
```

Comment:
I'd like to have this action and how it interacts with ofi_check_domain_attr/ofi_check_mr_mode documented to some extent.